packages/html/src/events/touch.rs

```diff
@@ -1,17 +1,147 @@
 use dioxus_core::Event;
 
 pub type TouchEvent = Event<TouchData>;
-#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
-#[derive(Debug, Clone, PartialEq, Eq)]
 pub struct TouchData {
-    pub alt_key: bool,
-    pub ctrl_key: bool,
-    pub meta_key: bool,
-    pub shift_key: bool,
-    // get_modifier_state: bool,
-    // changedTouches: DOMTouchList,
-    // targetTouches: DOMTouchList,
-    // touches: DOMTouchList,
+    inner: Box<dyn HasTouchData>,
+}
+
+impl<E: HasTouchData> From<E> for TouchData {
+    fn from(e: E) -> Self {
+        Self { inner: Box::new(e) }
+    }
+}
+
+impl std::fmt::Debug for TouchData {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("TouchData")
+            .field("alt_key", &self.alt_key())
```

Comment:
touch event can also use the InteractionModifiers trait

Reply:
Yes, I appreciate the review, but I'm still working on implementing touch events, so I wouldn't spend too much time reviewing this until I open the PR again